waku/protocol/v1/waku_protocol.nim

```diff
@@ -284,6 +289,8 @@ p2pProtocol Waku(version = wakuVersion,
         # await peer.disconnect(SubprotocolReason)
         continue
 
+      peer.state.accounting.received += 1
```

Comment:
I wonder if it doesn't make more sense to use existing metric abstractions, gauges etc. See Node Metrics. Would that not meet requirements?

Reply:
I was wondering the same thing a while back, which is actually why I haven't added this yet myself.

I think we can probably add the metric with a `peer` label, which would initially make us reach the goal of having the accounting per peer.
However, if we then want to use those numbers to e.g. rate peers in the protocol, we need proper access to those numbers. Not sure if metrics is intended to be used like that.
So the question is basically if we need some accounting underneath that reports those numbers to the metrics counters, or not.

Reply:
the `metrics`  that get collected by prometheus / written to `nim-metrics` are generally a one-way street - ie it's a pure output - if the application logic needs metrics for its own logic, it should define its own counters etc (and  possibly export these to nim-metrics, and perhaps call it something else to avoid the confusion.

Reply:
Yeah, exactly. Was thinking in the same direction.